File: src/invoicy-web/src/app/shared/models/address-model.ts
export interface AddressResponseModel {
  id: number;
  province: string;
  city: string;
  suburb?: string | null;
  street: string;
  zipCode: string;
}

export interface AddressRequestModel {
  province: string;
  city: string;
  suburb?: string | null;
  street: string;
  zipCode: string;
}

File: src/Invoicy.Application/Customers/Models/AddressModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Invoicy.Application.Customers.Models;

public class AddressModel
{
    public int Id { get; set; }
    public required string Province { get; set; }
    public required string City { get; set; }
    public string? Suburb { get; set; }
    public required string Street { get; set; }
    public required string ZipCode { get; set; }
}
